src/compiler/chainrules.jl

```diff
@@ -4,6 +4,7 @@
 @inline ZygoteRules.unthunk_tangent(x::NTuple{N,<:Number}) where N = x
 @inline ZygoteRules.unthunk_tangent(x::AbstractArray{<:Number,N}) where N = x
 @inline ZygoteRules.unthunk_tangent(x::AbstractArray) = map(unthunk_tangent, x)
+@inline ZygoteRules.unthunk_tangent(r::Base.RefValue) = r[] = unthunk_tangent(r[])
```

Comment:
Having second thoughts. Why is this mutating and not creating a new Ref?

Reply:
This is the same ref used in the gradient accumulation, creating a new ref would break the provenance that the ref was holding on. In case the ref needs to be accessed downstream from where the unthunking happens. It keeps a single source of truth.